src/pages/index.js

```diff
@@ -1,120 +1,267 @@
 /* --------------------- IMPORT CLASSES --------------------- */
 import "./index.css";
+import Api from "../components/Api.js";
 import Card from "../components/Card.js";
-import PopupWithImage from "../components/PopupWithImage.js";
-import PopupWithForm from "../components/PopupWithForm.js";
-import UserInfo from "../components/UserInfo.js";
 import FormValidator from "../components/FormValidator.js";
+import PopupWithForm from "../components/PopupWithForm.js";
+import PopupWithImage from "../components/PopupWithImage.js";
+import PopupConfirm from "../components/PopupConfirm.js";
 import Section from "../components/Section.js";
-import { initialCards, selectors, config } from "../utils/constants.js";
+import UserInfo from "../components/UserInfo.js";
+import { selectors, config } from "../utils/constants.js";
+
+const api = new Api({
+  baseUrl: "https://around.nomoreparties.co/v1/cohort-3-en",
+  headers: {
+    authorization: "0d34a92c-1da7-4adc-abc0-e2b3b38a4b1f",
+    "Content-Type": "application/json",
+  },
+});
 
-// Find Profile and Add Card buttons.
-const profileEditButton = document.querySelector("#profile-edit-button");
+// Find buttons.
+const editAvatButton = document.querySelector("#profile-avatar-container");
+const editProfButton = document.querySelector("#profile-edit-button");
 const addCardButton = document.querySelector("#add-card-button");
 
 // Find edit form input elements.
-const profileTitleInput = document.querySelector("#profile-title-input");
-const profileDescriptionInput = document.querySelector(
-  "#profile-description-input"
-);
+const profNameInput = document.querySelector("#profile-title-input");
+const profAboutInput = document.querySelector("#profile-about-input");
 
 //  Find form elements.
-const profileEditForm = document.forms["edit-card-form"];
-const addCardModalForm = document.forms["add-card-form"];
+const profEditForm = document.forms["edit-card-form"];
+const addCardForm = document.forms["add-card-form"];
+const avatEditForm = document.forms["avatar-edit-form"];
 
 /* ------------------- Form Validation -------------------- */
 
-const addFormValidator = new FormValidator(config, addCardModalForm);
-const editFormValidator = new FormValidator(config, profileEditForm);
+const addFormValidator = new FormValidator(config, addCardForm);
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object `formValidators`.  And then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
const formValidators = {}

// enable validation
const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // here you get the name of the form
    const formName = formElement.getAttribute('name')

   // here you store a validator by the `name` of the form
    formValidators[formName] = validator;
   validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```